### PR TITLE
chore(swap): Handle CcmFailed event in status [WEB-333]

### PR DESCRIPTION
**Pull request by niklasnatter**

### CODE REVIEW

File: packages/swap/src/routes/v2/utils.ts [hunk 1]
```diff
@@ -81,7 +81,8 @@ export const getLatestSwapForId = async (id: string) => {
     }
 
     swapRequest = swapDepositChannel.swapRequests.at(0);
-    if (!swapRequest) {
+    // swap request will have no swaps for failed ccm swaps
+    if (!swapRequest || !swapRequest.swaps.length) {
```
[the remark this answers is not in the document]
Comment (acdibble) [reply packages/swap/src/routes/v2/utils.ts:85]: not sure what the actual property is called

File: packages/swap/src/routes/__tests__/v2/swap.test.ts [hunk 2]
```diff
@@ -702,6 +734,96 @@ describe('server', () => {
...
+          swapEventMap['Swapping.CcmFailed'],
+          finalizedEvent,
+          swapEventMap['Swapping.SwapRequestCompleted'],
+        ],
+        [
+          {
+            id: '0000000092-000399-02fea',
+            call: {
+              value: {
+                __kind: 'process_deposits',
```
Comment (zoheb391) [inline packages/swap/src/routes/__tests__/v2/swap.test.ts:789]: what exactly is this event ? 🧐

**Review by zoheb391 — APPROVED** [review]

File: packages/swap/src/routes/v2/utils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (niklasnatter) [reply packages/swap/src/routes/v2/utils.ts:85]: makes sense!

File: packages/swap/src/routes/__tests__/v2/swap.test.ts (hunk 2 above)
Comment (niklasnatter) [reply packages/swap/src/routes/__tests__/v2/swap.test.ts:789]: its not an event, but a call - the `ccmFailed` handler fetches this from the indexer to get some additional information: https://github.com/chainflip-io/chainflip-sdk-monorepo/blob/7d97a8691fa14243ae1f234e05b66e4610b96897/packages/swap/src/event-handlers/ccmFailed.ts#L46-L48

**Review by jerryafr — APPROVED** [review]

**Pull request merged**